### DISCUSSION TITLE
v1.0.2

### DISCUSSION THREAD

**Discussion by AndyFug** [bitfocus/companion-module-bss-soundweb#10] [Releases]
Fixes and minor improvements.

- Fixes bug where large relative percent increments could result in looping back round to 0%
- Minor refactoring for logging module TX messages when setting parameters to avoid code duplication and error

<hr /><em>This discussion was created from the release <a href='https://github.com/bitfocus/companion-module-bss-soundweb/releases/tag/v1.0.2'>v1.0.2</a>.</em>

**Comment by joeltcaron** [comment]
Just wanted to say thanks to those who have contributed to this project. Although what I was doing was very simple, I was able to read the documentation on finding the node addresses and was able to make companion and a BLU160 talk to each other with DSP status/ feedback. Thanks again!

**Reply by AndyFug** [reply]
Thanks for the positive feedback.  Really pleased to hear the module is being used.

**Comment by FLB58** [comment]
Hi Andy,
great work! I just discovered your module today. I have been programming BLU DSPs since 2018. Last year I discovered the Stream Deck and have been using the generic TCP module to send hex strings to the BLU-100. I am alright with the addresses and parameters in your module, but I am struggling with setting the feedback section such that I can get the dB value of a gain object from the DSP. I am using Companion version 4.03 and I can't seem to find the the 'Create variable?' option in the feedback section you are mentioning under "Parameter Values". Also, what is meant with "Variable tag (ID)"? May I kindly ask you to enlighten me?
Screenshot: [temp.pdf](https://github.com/user-attachments/files/22242425/temp.pdf)
Thanks and regards,
Frank

**Reply by AndyFug** [reply]
Hi.

So by looking at your PDF, it appears that you are using the `Add Parameter Variable` feedback.  Because this feedback is designed to be used purely to add a variable, it doesn't have the `Create Variable?` tick box, because its one job in the world is to create a variable. :). The `Create Variable?` tick box is available on other feedbacks as a covenience so that you don't have to add a separate `Add Parameter Variable` feedback just to create a variable when you have an existing feedback setup for the given parameter.

To add your dB value to the button's text, you have two options:

1. Navigate to the "Variables" section in Companion (on the left navigation bar in V4) and find the variables associated with the Soundweb connection you have setup.  You should find in the list of variables your gain parameter that you setup in your feedback.  Copy the variable, which should look a little like `$(BSS_Soundweb:param_0.3.0.0.0.0-DB)` and is in red text.  (Note there is a little copy shortcut/symbol after the text).  You can paste that variable into your button text, and it should show the current dB value of the gain parameter you are referencing.

2. Start typing something like `$(Soun` straight into your button text field and it should then show a dropdown list of the Soundweb variables (as a shortcut).  You should then be able to pick your variable from that list and it auto-complete the variable name.

To help identify a specific variable when you have a long list of them, I introduced the `Variable Tag` property, just so you can add a friendly name to it.  For instnace, you might provide a tag called `myCh1Gain` or something.  Once you've done this, you'll see this tag reflected in the variable name.  This means when you start typing a variable into your text field as described in option 2 above, you can type something like `$(myCh1G` and it will find the variable you have created more easily.  If you don't provide a tag, the only way you can identify your variable is by looking for its parameter address, which is obviously not very human friendly.

I hope that helps... sorry I can't provide more screenshots etc at the moment.  If you're not too familiar with variables, it may be worth checking the Companion help docs.  Variables really do open up a whole other powerful side to Companion... especially when you combine them with expressions.

Andy

**Comment by FLB58** [comment]
Hi Andy,

thanks for the info you provided. I started a new config in my BLU-100 and a new companion config as well, so I don’t get any weird side effects. I got the gain object working, including getting feedback into the variable. The mute object works only with n-channel gain objects, while the gain object also works with single gains objects. I will need to change my design a bit. The mute object does not work with n-channel gain if you add a parameter value to the object address. The parameter bit appears to be added behind the scenes from selecting the right channel. Which is when Mute started to work.

Either way, using this module helps to reduce the complexity of my BLU config quite a bit. I was using a pulse in conjunction with a logic source linked to bump up/bump down objects, which adds up when you do that for 8 channels.

Thanks again!

Regards,

Frank

**Comment by FLB58** [comment]
Hi Andy,

with Companion release 4.1, connection collections were added. However, BSS_Soundweb does not support connection collections, which limits the efficiency otherwise gained by using a connection collection. Are you planning an update that will support connection collections?

An update on controlling Mute of a single-channel gain object: it works beautifully with your Generic Button.

Regards,

Frank

 

From: ***@***.*** ***@***.***> 
Sent: Mittwoch, 10. September 2025 22:43
To: 'bitfocus/companion-module-bss-soundweb' ***@***.***>
Subject: RE: [bitfocus/companion-module-bss-soundweb] v1.0.2 (Discussion #10)

 

Hi Andy,

thanks for the info you provided. I started a new config in my BLU-100 and a new companion config as well, so I don’t get any weird side effects. I got the gain object working, including getting feedback into the variable. The mute object works only with n-channel gain objects, while the gain object also works with single gains objects. I will need to change my design a bit. The mute object does not work with n-channel gain if you add a parameter value to the object address. The parameter bit appears to be added behind the scenes from selecting the right channel. Which is when Mute started to work.

Either way, using this module helps to reduce the complexity of my BLU config quite a bit. I was using a pulse in conjunction with a logic source linked to bump up/bump down objects, which adds up when you do that for 8 channels.

Thanks again!

Regards,

Frank

**Comment by AndyFug** [comment]
Hi,

I haven't been using Companion for a few months so I haven't looked at the new features in v4 properly yet.  I've had a quick look at the release notes for v4.1 and the 'connection collections' feature doesn't look like something that would be dependent on module support... Ie... It looks like a feature with the Companion application itself rather than the module.  

When I get some time, I will look into it.  There are some other features in 4.1 that look like they will require updates in the module for support.

A feature I do have in the pipeline already is adding support for meters... I just need to complete testing/documentation on that before I release it.
